REST/src/main/java/pl/jakubowskiprzemyslaw/REST/services/FleetGeneratorService.java

```diff
@@ -0,0 +1,140 @@
+package pl.jakubowskiprzemyslaw.REST.services;
+
+import pl.jakubowskiprzemyslaw.REST.models.EmptyFields;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.Fleet;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.Mast;
+import pl.jakubowskiprzemyslaw.tajgertim.models.board.Ship;
+import pl.jakubowskiprzemyslaw.tajgertim.models.configuration.ship.ShipAmount;
+import pl.jakubowskiprzemyslaw.tajgertim.models.configuration.ship.ShipConfiguration;
+import pl.jakubowskiprzemyslaw.tajgertim.models.configuration.ship.ShipSize;
+import pl.jakubowskiprzemyslaw.tajgertim.models.coordinates.Coordinate;
+
+
+import java.util.*;
+
+public class FleetGeneratorService {
+
+    private ShipConfiguration shipConfiguration;
+    private int boardSize;
+    private EmptyFields emptyFields;
+
+    public FleetGeneratorService() {
+        generateDefaultShipConfiguration();
+    }
+
+    private void generateDefaultShipConfiguration() { //only for testing
+        shipConfiguration = new ShipConfiguration();
+        shipConfiguration.put(ShipSize.MAST_4, new ShipAmount("1"));
+        shipConfiguration.put(ShipSize.MAST_3, new ShipAmount("2"));
+        shipConfiguration.put(ShipSize.MAST_2, new ShipAmount("3"));
+        shipConfiguration.put(ShipSize.MAST_1, new ShipAmount("4"));
+
+        boardSize = shipConfiguration.getBoardSize();
+    }
+
+    public Fleet generateRandomFleet() {
+        emptyFields = new EmptyFields(boardSize);
+
+        Fleet fleet = new Fleet();
+        List<Integer> shipPlacementList = shipConfiguration.getShipPlacementList();
+
+        for (Integer shipLength : shipPlacementList) {
+            Ship ship = placeShip(shipLength);
+            fleet.addShip(ship);
+            removeShipAndBufferFromEmptyFields(ship);
+        }
+        return fleet;
+    }
+
+    private Ship placeShip(Integer shipLength) {
+        Optional<Ship> ship = Optional.empty();
+
+        while (!ship.isPresent()) {
+            Coordinate randomCoordinate = emptyFields.getRandomCoordinate();
+            ship = tryToPlaceShip(shipLength, randomCoordinate);
+        }
+
+        return ship.get();
+    }
+
+    private Optional<Ship> tryToPlaceShip(int shipLength, Coordinate coordinate) {
+
+        List<Coordinate> shipCoordinates;
+        try {
+            shipCoordinates = tryToAllocateCoordinatesToBoard(shipLength, coordinate);
+        } catch (CannotAllocatePositionsOnBoardException cannotAllocatePositionsOnBoardException) {
+            return Optional.empty();
+        }
+
+        Ship ship = createShip(shipCoordinates);
+
+        return Optional.of(ship);
+    }
+
+    private List<Coordinate> tryToAllocateCoordinatesToBoard(int shipLength, Coordinate coordinate) throws CannotAllocatePositionsOnBoardException {
+        List<Coordinate> shipCoordinates = createShipCoordinatesWithFirstCoordinateAdded(coordinate);
```

Comment:
I would think about refactor for this method in the future. For now i can't suggest any good way in which this method should be changed.

Reply:
Will be considered later